generator/api/src/main/java/io/fabric8/maven/generator/api/support/AbstractPortsExtractor.java

```diff
@@ -0,0 +1,178 @@
+package io.fabric8.maven.generator.api.support;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import org.apache.maven.project.MavenProject;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+import io.fabric8.maven.core.util.PrefixedLogger;
+import io.fabric8.maven.generator.api.PortsExtractor;
+import io.fabric8.utils.Strings;
+
+public abstract class AbstractPortsExtractor implements PortsExtractor {
+
+    private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+    private static final ObjectMapper YAML_MAPPER = new ObjectMapper(new YAMLFactory());
+
+    private static final String DOT = ".";
+    private static final String JSON_EXTENSION = ".json";
+    private static final String YAML_EXTENSION = ".yaml";
+    private static final String YML_EXTENSION = ".yml";
+    private static final String PROPERTIES_EXTENSION = ".properties";
+
+    private static final String NUMBER_REGEX = "\\d+";
+
+    private static final String PORT_REGEX = "([a-zA-Z0-9_]+)(([\\.-_]+p{1})|([P]{1}))ort";
+    private static final Pattern PORT_PATTERN = Pattern.compile(PORT_REGEX);
+
+    private final PrefixedLogger log;
+
+    public AbstractPortsExtractor(PrefixedLogger log) {
+        this.log = log;
+    }
+
+    /**
+     * @return The name of the system property that points to the path of the configuration file.
+     */
+    public abstract String getConfigPathPropertyName();
+
+    /**
+     * Finds the name of the configuration file from the {@link MavenProject}.
+     * @param project   The {@link MavenProject} to use.
+     * @return          The path to the configuration file.
+     */
+    public abstract String getConfigPathFromProject(MavenProject project);
+
+
+    public File getConfigLocation(MavenProject project) {
+        //The system property has priority over what is specified in the pom.
+        String configPath = System.getProperty(getConfigPathPropertyName(), getConfigPathFromProject(project));
```

Comment:
Doesn't `System.getProperty()` throws an NPE on a null key ? Don't know by heart, but maybe that should checked before ?